reproduce-figures/prepare-scpca-portal-data.R

```diff
@@ -0,0 +1,309 @@
+      
+      # Define all sample directories present in this project, _excluding_ multiplexed samples
+      # This is because multiplexed samples aren't used in any figures created from ScPCA data files
+      sample_dirs <- list.dirs(
+        project_consensus_dir, 
+        full.names = TRUE
+      ) |>
+        purrr::set_names(\(x) {basename(x)}) |>
+        # get rid of the project folder & multiplexed samples
+        purrr::discard_at(\(x) {stringr::str_detect(x, "SCPCP\\d{6}")}) |>
+        purrr::discard_at(\(x) {stringr::str_detect(x, "_")}) 
+
+      ####### Step 1: Copy RDS files to target directories #######
+      sample_dirs |>
+        purrr::iwalk(
+          \(sample_dir, sample_id) {
+            
+            # Copy processed rds files to celltype_results_files
+            # The next step will parse these files into TSVs
+            if (sample_id %in% celltype_results_samples) {
+              system(glue::glue("cp {sample_dir}/*processed.rds {output_dir_celltype_results}"))
+            } 
+            
+            # Copy rds files to standalone sample dir
+            if (sample_id %in% standalone_sample_dirs) {
+              system(glue::glue("cp -r {sample_dir} {output_dir_s3_files}"))
+            } 
+          
+            # We can now remove the (un)filtered files
+            system(glue::glue("rm {sample_dir}/*filtered.rds"))
+      })
+
+      # If this project is used in the bulk analysis,copy project over as well
+      if (project_id %in% bulk_projects) {
+        fs::dir_copy(
+          project_consensus_dir, 
+          output_dir_analysis_data
+        )
+      }
+      
+      ####### Step 2: Prepare consensus cell type files #######
+      sample_dirs |>
+        purrr::walk(
+          \(sample_dir) {
+              
+            # Define input and output files
+            processed_rds_file <- list.files(
+              path = sample_dir,
+              pattern = "_processed\\.rds$", 
+              full.names = TRUE
+            )
+            stopifnot("Expected a single processed rds file." = length(processed_rds_file) == 1)
+            
+            # Define output file names
+            output_consensus_tsv <- stringr::str_replace(
+              processed_rds_file, "_processed\\.rds$", "_processed_consensus-cell-types.tsv.gz"
+            )
+      
+            output_markers_tsv <- stringr::str_replace(
+              processed_rds_file, "_processed\\.rds$", "_processed_marker-gene-expression.tsv.gz"
+            )    
+            
+            # Read in the processed SCE
+            sce <- readRDS(processed_rds_file)
+            
+            # Prepare and export the consensus cell types table
+            data.frame(
+              project_id = metadata(sce)$project_id, 
+              sample_id  = metadata(sce)$sample_id, 
+              library_id = metadata(sce)$library_id, 
+              sample_type = metadata(sce)$sample_type,
```

Comment:
```suggestion
              sample_type = paste0(metadata(sce)$sample_type, collapse = ",")
```
This is going to cause you issues as is for the samples that have multiple sample types.